synthesizer/src/vm/finalize.rs

```diff
@@ -716,6 +716,25 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                         finalize_operations.push(operation);
                     }
 
+                    // Iterate over the bonded balances.
+                    for (address, amount) in bonded_balances {
+                        // Construct the key.
+                        let key = Plaintext::from(Literal::Address(*address));
+                        // Retrieve the current bonded balance.
+                        let value = store.get_value_speculative(program_id, bonded_mapping, &key)?;
+                        // Compute the next bonded balance.
+                        let next_value = Value::from(Literal::U64(U64::new(match value {
+                            Some(Value::Plaintext(Plaintext::Literal(Literal::U64(value), _))) => {
+                                (*value).saturating_add(*amount)
+                            }
+                            None => *amount,
+                            v => bail!("Critical bug in pre-ratify - Invalid bonded balance type ({v:?})"),
+                        })));
+                        // Update the bonded balance in finalize storage.
+                        let operation = store.update_key_value(program_id, bonded_mapping, key, next_value)?;
+                        finalize_operations.push(operation);
+                    }
```

Comment:
I haven't looked into this in depth, but outside of the `bonded` mapping, we need to update the `committee_state` in the `committee` mapping and also the number of delegators if/when this PR gets merged in - https://github.com/AleoHQ/snarkVM/pull/2213.

In addition, we need checks that the delegated bonding amount meets all the requirements (for example, delegation must be more than X credits)